tests/ResidueSplitTest.php

```diff
@@ -14,286 +14,458 @@
 
-        if (null !== $expectedStepRemainder) {
-            $this->assertEquals($expectedStepRemainder, $residue->getStepRemainder());
-        }
+        $this->assertEquals($expectedremainder, $residue->getRemainder());
     }
 
-    public function dataProviderStepValues()
+    public function dataProviderEquityValues()
     {
-        yield '100/3' => [
-            100, // value
-            3, // divide
-            2, // decimal value
-            0.05, // step
-            [
-                33.35,
-                33.35,
-                33.30,
-            ], // expected result
-        ];
-
-        yield '0.15/4' => [
-            0.15, // value
-            4, // divide
-            2, // decimal value
-            0.05, // step
+        yield '1/1 - Step 0.0' => [
+            1, // value
+            1, // divide
+            null, // decimal value
+            null, // step
             [
-                0.05,
-                0.05,
-                0.05,
-                0.00,
+                1,
             ], // expected result
+            0, // reminder
         ];
 
-        yield '12.25/2' => [
-            12.25, // value
-            2, // divide
-            2, // decimal value
-            0.05, // step
-            [
-                6.15,
-                6.10,
-            ], // expected result
-        ];
-
-        yield '7.315/3 - 0.005' => [
-            7.315, // value
-            3, // divide
-            3, // decimal value
-            0.005, // step
-            [
-                2.440,
-                2.440,
-                2.435,
-            ], // expected result
-        ];
-
-        yield '7.315/3 - 0.05' => [
-            7.315, // value
+        yield '100/3 - Step 0.0' => [
+            100, // value
             3, // divide
-            3, // decimal value
-            0.05, // step
+            0, // decimal value
+            null, // step
             [
-                2.45,
-                2.45,
-                2.40,
+                33,
+                33,
+                33,
             ], // expected result
-            0.015, // expected remainder
+            1, // reminder
         ];
 
-        yield '99.99/3 - 0' => [
-            99.99, // value
+        yield '100/3 - Step 0.01' => [
+            100, // value
             3, // divide
             2, // decimal value
-            0, // step
+            null, // step
             [
                 33.33,
                 33.33,
                 33.33,
             ], // expected result
-            0, // expected remainder
+            0.01, // reminder
+        ];
+
+        yield '2.91/2 - Step 0.1' => [
+            2.91, // value
+            2, // divide
+            null, // decimal value
+            0.03, // step
+            [
+                1.44,
+                1.44,
+            ], // expected result
+            0.03, // reminder
         ];
     }
```

Comment:
do you think it's relevant to add a this test:

```php

    /**
     * @test
     */
    public function it_should_split_returns_iterator(): void
    {
        $residue = Residue::create(100)->divideBy(3);

        $this->assertSame([33.34, 33.33, 33.33], iterator_to_array($residue->split(), false));
    }
```

Reply:
I'm not sure to see the interest here because this is the behavior of toArray which is called for each test...

It seems to me that it's a bit like testing the behavior of the iterator_to_array function ...

The question is, can we write a test that would allow to fail the expected result with iterator_to_array ?

Reply:
You're right, testing split is not relevant here _(split() output is typed)_.

Another proposition on the tests, would we not test the returned total? _(sum values + remainder)_:

```php
$this->assertEquals($value, array_sum($residue->toArray()) + $residue->getRemainder());
```

Reply:
> Another proposition on the tests, would we not test the returned total? _(sum values + remainder)_:
> 
> ```
> $this->assertEquals($value, array_sum($residue->toArray()) + $residue->getRemainder());
> ```

Just great! It freezes the box perfectly, no chance of leakage while writing the tests.